src/unit.rs

```diff
@@ -908,13 +910,33 @@ pub enum AttributeValue<'input, Endian>
     /// A slice of an arbitrary number of bytes.
     Block(EndianBuf<'input, Endian>),
 
-    /// A one, two, four, or eight byte constant data value. How to interpret
-    /// the bytes depends on context.
+    /// A one byte constant data value. How to interpret the byte depends on context.
     ///
     /// From section 7 of the standard: "Depending on context, it may be a
     /// signed integer, an unsigned integer, a floating-point constant, or
     /// anything else."
-    Data(EndianBuf<'input, Endian>),
+    Data1([u8; 1]),
+
+    /// A two byte constant data value. How to interpret the bytes depends on context.
+    ///
+    /// From section 7 of the standard: "Depending on context, it may be a
+    /// signed integer, an unsigned integer, a floating-point constant, or
+    /// anything else."
+    Data2([u8; 2]),
+
+    /// A four byte constant data value. How to interpret the bytes depends on context.
+    ///
+    /// From section 7 of the standard: "Depending on context, it may be a
+    /// signed integer, an unsigned integer, a floating-point constant, or
+    /// anything else."
+    Data4([u8; 4]),
+
+    /// An eight byte constant data value. How to interpret the bytes depends on context.
+    ///
+    /// From section 7 of the standard: "Depending on context, it may be a
+    /// signed integer, an unsigned integer, a floating-point constant, or
+    /// anything else."
+    Data8([u8; 8]),
```

Comment:
I don't see how we can `repr(u64)` *and* and have an eight byte sized variant *and* have room for the discriminator. The size of `AttributeValue` is surely larger than `8`, no? Am i missing something? Doing poor arithmetic?

That is, with this second commit, is `repr(u64)` still doing anything?